xclim/ensembles/_filters.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import numpy as np
+import xarray as xr
+
+
+def _concat_hist(da, **hist):
+    """Concatenate historical scenario with future scenarios along time.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Input data where the historical scenario is stored alongside other, future, scenarios.
+    hist: {str: str}
+      Mapping of the scenario dimension name to the historical scenario coordinate, e.g. `scenario="historical"`.
+
+    Returns
+    -------
+    xr.DataArray
+      Data with the historical scenario is stacked in time before each one of the other scenarios.
+
+    Notes
+    -----
+    Data goes from:
+
+        +------------+----------------------------------+
+        | scenario   | time                             |
+        +============+==================================+
+        | historical | hhhhhhhhhhhhhhhh                 |
+        +------------+----------------------------------+
+        | ssp245     |                 111111111111     |
+        +------------+----------------------------------+
+        | ssp370     |                 222222222222     |
+        +------------+----------------------------------+
+
+    to:
+        +----------+----------------------------------+
```

Comment:
Something strange going on with the formatting here: 
![image](https://github.com/Ouranosinc/xclim/assets/10819524/aab0bdda-f37c-472c-80bb-a471a39364cc)